src/opal.c

```diff
@@ -905,6 +905,15 @@ get_identifier_lexeme (int char_line, int char_col)
   /// Terminate string
   identifier_str[str_len++] = '\0';
 
+  /// Error for unsupported characters
+  if (str_len == 1)
+    {
+      errno = EXIT_FAILURE;
+      perror ("Unsupported character");
```

Comment:
Macro `EXIT_FAILURE` expands to 1. This means your system error message will be:
`Unsupported character: Operation not permitted`

Also, errors not originating from a system call failure, should not be printed with perror().
I suggest something like this: 

`logger(ERROR, "[%d: %d] Invalid identifier.", char_line, char_col);`